pkg/services/ngalert/state/state.go

```diff
@@ -297,10 +298,10 @@ func resultError(state *State, rule *models.AlertRule, result eval.Result, logge
 				state.Annotations["Error"] = result.Error.Error()
 				// If the evaluation failed because a query returned an error then add the Ref ID and
 				// Datasource UID as labels
-				var queryError expr.QueryError
+				var queryError = expr.MakeQueryError("", "", nil).(errutil.Error)
```

Comment:
The lint error is correct, if a little bit excessive in your case, since you know it won't be wrapped.
But since you're overriding this with `errors.As` in the next line, maybe you only need

```suggestion
				var queryError errutil.Error
```

here?

Reply:
@emil Will this match the specific base in errors.go (`QueryError`) only or any errutil.Error ?

Reply:
Any errutil.Error, but that's true for what you already have as well, [`errors.As`](https://pkg.go.dev/errors#As) without a custom `As(any) bool` method doesn't perform any equivalence checks. I believe what you'd need here is something like:

```
if errors.As(state.Error, &queryError) && errors.Is(queryError, expr.QueryError) {
```

(or keeping the old logic entirely and have it create an `errutil.Error` on unwrap)

Reply:
Confused since QueryError is a errutil.Template ...

Reply:
For `Is` specifically, there is an override that lets you determine whether the error stems from a given Base/Template using `errors.Is`, see https://github.com/grafana/grafana/blob/9b6c7b74c7d554e3ed682cf0341f147e7b3d72c5/pkg/util/errutil/errors.go#L218-L219